subsys/bluetooth/host/l2cap.c

```diff
@@ -2611,7 +2613,13 @@ static void l2cap_chan_le_recv(struct bt_l2cap_le_chan *chan,
 		return;
 	}
 
-	err = chan->chan.ops->recv(&chan->chan, buf);
+	owned_ref = net_buf_ref(buf);
+	err = chan->chan.ops->recv(&chan->chan, owned_ref);
```

Comment:
What if `chan->chan.ops->recv == NULL`?

Reply:
Out of scope of this PR?

Reply:
Trivial fix though - Just apply the same solution as we have for `alloc_buf` on line 2589.

Reply:
But unlike `alloc_buf`, `recv` is mandatory. The only way for this to fail is if the application messes with the `ops` struct when it shouldn't. That's a serious error that might as well trigger a segfault. There is no bug here.

We could be nice and recover. But there are many ways. Just substitute in a no-op `recv`, like you say? But then the application does not know it broke something. Disconnect the channel?

Defensive code is nice, but I would prefer this is discussed in a separate issue. Then @jori-nordic and @jhedberg can give their input. Will you make an issue @Thalley?

Reply:
> But unlike alloc_buf, recv is mandatory.

Where is that documented? Implied by the lack of `if` in the description?

> The only way for this to fail is if the application messes with the ops struct when it shouldn't. That's a serious error that might as well trigger a segfault. There is no bug here.

Why shouldn't the application modify this? It's application owned, isn't it? 

> Just substitute in a no-op recv, like you say? But then the application does not know it broke something. 

Not sure if I would consider that broken. It's (IMO) not much different than not providing the `recv` callback for scanning. 

I'm not understanding the reluctance of checking if a pointer is NULL before dereferencing it

Reply:
> here my good sir

I forgot about that. I was too busy looking in the documentation which doesn't describe this requirement (it probably should, but that's for another PR). 

> An __ASSERT_NO_MSG() maybe, but I wouldn't add error handling.

Since this would be due to an application, I don't think an assert is the right course here. 
`LOG_ERR` maybe? Failing on segmentation fault just seem like a subpar approach

Reply:
The problem is that error logs are not actionable by said app. So say it malloc'd the ops struct and accidentally frees it after zeroing it out. It won't know that the channel is still alive and receiving data.

I agree that this situation really sucks because asserts are disabled in prod. So the app won't know in either case. Maybe a `BT_ASSERT()`?

Reply:
@Thalley https://github.com/zephyrproject-rtos/zephyr/pull/77115

Reply:
> The problem is that error logs are not actionable by said app.

Well neither are asserts, right? Both can give information to the developer, but the app itself cannot do anything, so in that regard they are equally good, no? 

Of course NULL checks won't fix all memory issues either as `recv = 0x1234` is non-NULL but would still break stuff. 

At this point I am OK with any solution, but not checking an application provided pointer before dereferencing is not great :)